pkgs/zphinxzerver/default.nix

```diff
@@ -0,0 +1,38 @@
+{ pkgs, version, src, bearssl-src, equihash, libsphinx-src, zigtoml-src }:
+with pkgs;
+stdenv.mkDerivation {
+  pname = "zphinxzerver";
+  inherit version src;
+
+  buildInputs = [ libsodium.dev rename zig ];
+
+  postPatch = ''
+    substituteInPlace build.zig \
+      --replace 'exe.addIncludeDir(".");' \
+      'exe.addIncludeDir("."); exe.addIncludeDir("${libsodium.dev}/include"); exe.addIncludeDir("${equihash}/include");'
+
+    substituteInPlace build.zig \
+      --replace 'exe.addLibPath("/usr/lib");' \
+      'exe.addLibPath("/usr/lib"); exe.addLibPath("${equihash}/lib");'
```

Comment:
I'm not so happy with this patching here. It feels somewhat ugly. Here is the [original file](https://github.com/stef/zphinx-zerver/blob/master/build.zig). We need to make sure that the zig compiler finds our libsodium & equihash libs.

Reply:
I assume what is ugly here is relying on arbitrary strings (`exe.addIncludeDir(".");`) right?
For that one can use patch files to insert anchors which would then be replaced with substituteInPlace